umbraco/Umbraco.CMS.Backoffice
# Feature/localization part 1

## Description

This is step 1 of the RFC for localization: https://github.com/umbraco/Umbraco.CMS.Backoffice/discussions/818

**Storybook:** [There is a new documentation page which you can see here](https://ambitious-stone-0033b3603-817.westeurope.1.azurestaticapps.net/?path=/docs/guides-localization--docs).

## What changed

**New element: umb-localize**
```javascript
render() {
  return html`<umb-localize key="general_close"></umb-localize>`
}
```

**New context: UmbLocalizationContext**
Has two methods `localize` and `localizeMany` that return an observable with the localized values of the requested keys.

## Types of changes


## Motivation and context

This allows UI developers to add localized strings to their UI, which is one of the success criteria for migrating to the new Backoffice.

## How to test?

Add a new UI element (or amend an existing one) and test by one of these scenarios:

**Test the render function (should render "Close" in the browser)**
```javascript
render() {
  return html`<umb-localize key="general_close"></umb-localize>`
}
```

**Test the UmbLocalizeController**
```javascript
render() {
  return html`<uui-button .label=${this.localize.term('general_close')}></uui-button>`
}
```

## Screenshots

Changing the user language updates a few strings:

https://github.com/umbraco/Umbraco.CMS.Backoffice/assets/752371/d7f775d7-b805-433d-943b-c70b7351d4e1



## Checklist
